### ISSUE TITLE
Bug in Frontend::prepareMetaDescription()

### ISSUE THREAD

**Issue by Toflar** [contao/core#8030]
My customer uses the InsertTag `{{file::*}}` to override the `<meta name="description" ...>`. Now the problem is that this InsertTag returns `<img>` and other HTML tags with attributes (though containing `"`) which obviously causes trouble, which is why `Frontend::prepareMetaDescription()` calls `strip_tags()`.

However, as `file` is one of the InsertTags that is not replaced when caching is enabled, this content will not get stripped. So I'm not sure what we should do here. Either we have to do this:

```diff
--- a/system/modules/core/classes/Frontend.php
+++ b/system/modules/core/classes/Frontend.php
@@ -622,7 +622,7 @@ abstract class Frontend extends \Controller
 	 */
 	protected function prepareMetaDescription($strText)
 	{
-		$strText = $this->replaceInsertTags($strText);
+		$strText = $this->replaceInsertTags($strText, false);
 		$strText = strip_tags($strText);
 		$strText = str_replace("\n", ' ', $strText);
 		$strText = \StringUtil::substr($strText, 180);
```

Or we have to make sure we strip any non-cachable InsertTags out of the meta description.

What do you think?

**Comment by leofeyer** [comment]
As discussed on Mumble on October 14th, adding `false` is the correct solution.

**Comment by leofeyer** [comment]
Fixed in 7e8489a7762128d3c2e9678f8a17b1decaebacd0.
